### PR TITLE
Feature/create proposal freelancer

### PR DESCRIPTION
**Pull request by KseniaKoms**
add freelancer`s proposal module
add storing files logic

### CODE REVIEW

File: src/modules/proposals/services/proposals.service.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    file: Express.Multer.File,
+  ): Promise<Proposal> {
+    try {
+      const filePath = file.path;
+
+      const proposal = await this.proposalRepository.create({
+        ...createProposalDto,
+        filePath: filePath,
+      });
+      await this.proposalRepository.save(proposal);
```
Comment (lyudad) [inline src/modules/proposals/services/proposals.service.ts]: why we need create and then save? why we can make save in one commend?

Comment (KseniaKoms) [reply src/modules/proposals/services/proposals.service.ts]: updated

File: src/utils/fileName.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { extname } from 'path';
+
+export const fileName = (_req, file, callback) => {
+  const uniqueSuffix = Date.now() + '-' + Math.round(Math.random() * 1e9);
+  const ext = extname(file.originalname);
+  const filename = `${uniqueSuffix}${ext}`;
+
+  callback(null, filename);
+};
```
Comment (lyudad) [inline src/utils/fileName.ts:14]: look like you took it from internet and its fine, but you still need to add types

Comment (KseniaKoms) [reply src/utils/fileName.ts:14]: updated

**Review by lyudad — APPROVED** [review]

**Review by lyudad — APPROVED** [review]

**Pull request merged**